File: src/net/dinkla/raytracer/math/Ray.java
package net.dinkla.raytracer.math;

/**
 * Created by IntelliJ IDEA.
 * User: jorndinkla
 * Date: 10.04.2010
 * Time: 15:28:13
 * To change this template use File | Settings | File Templates.
 */
public class Ray {

    public final Point3DF o;
    public final Vector3DF d;

    public Ray(final Point3DF origin, final Vector3DF direction) {
        this.o = origin;
        this.d = direction;
    }

    public Ray(Ray ray) {
        this.o = ray.o;
        this.d = ray.d;
    }

    public Point3DF linear(final float t) {
        return o.plus(d.mult(t));
    }

    @Override
    public String toString() {
        return "Ray(" + o.toString() + ", " + d.toString() + ")";
    }

}

File: src/net/dinkla/raytracer/math/Ray.scala
package net.dinkla.raytracer.math

/**
 * Created by Dinkla on 10.07.2015.
 */
class Ray(val o: Point3DF, val d: Vector3DF) {

  def this(ray: Ray) = this(ray.o, ray.d)

  def linear(t: Float): Point3DF = o + (d * t)

  override def toString: String = s"Ray($o, $d)"

  def getO = o
  def getD = d

}

/*

later:

class Ray[F](val o: Point3D[F], val d: Vector3D[F]) {

  def this(ray: Ray) = this(ray.o, ray.d)

  def linear(t: F): Point3D[F] = o + (d * t)

  override def toString: String = s"Ray($o, $d)"
}

*/

